### PR TITLE
Remove checks for long-discontinued applications

### PR DESCRIPTION
**Pull request by LumioseSil**
#### f488ed361cf13d4c8d976cd33af22609b4f340f2
<pre>
Remove checks for long-discontinued applications
<a href="https://bugs.webkit.org/show_bug.cgi?id=251706">https://bugs.webkit.org/show_bug.cgi?id=251706</a>

&lt;rdar://problem/4985321&gt;
&lt;rdar://problem/5009625&gt;

Reviewed by Brent Fulgham.

Aperture is long gone; the program no longer runs on supported macOS
releases. We can remove the checks for Aperture.

Same as iTunes and AOL on Mac.

Finally, remove the iBooks check since iBooks is also no longer affected by its old quirk.

*Source/WebCore/Modules/geolocation/Geolocation.cpp:
*Source/WebCore/WebCore.order:
*Source/WebCore/platform/RuntimeApplicationChecks.h:
*Source/WebCore/platform/cocoa/RuntimeApplicationChecksCocoa.mm:
*Source/WebKitLegacy/mac/Misc/WebKitVersionChecks.h:
*Source/WebKitLegacy/mac/WebCoreSupport/WebEditorClient.mm:
*Source/WebKitLegacy/mac/WebView/WebImmediateActionController.mm:
*Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm:
*Source/WebKitLegacy/mac/WebView/WebView.mm:

Canonical link: <a href="https://commits.webkit.org/259988@main">https://commits.webkit.org/259988@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/ae3dd7e13d9f17572cb6ebe05a091523faba2a32

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106623 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15620 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39415 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115808 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/115428 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110530 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17128 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6853 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98815 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112392 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/13008 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/95976 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/40586 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/94886 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27621 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/82315 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8846 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28974 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9398 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/6014 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15013 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48521 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6904 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10926 "Built successfully") | | | 
| | | | |

### CODE REVIEW

**Review by karlcow — APPROVED** [review]
LGTM. but let's ask someone from @brentfulgham team too.

**Review by marcoscaceres — COMMENTED** [review]
Probably want to someone Apple internal to verify all the radars are closed or otherwise resolved, etc. 

Has that been done?

File: Source/WebCore/platform/cocoa/RuntimeApplicationChecksCocoa.mm [hunk 1]
```diff
@@ -188,12 +176,6 @@ static bool applicationBundleIsEqualTo(const String& bundleIdentifierString)
     return isQuickenEssentials;
 }
 
-bool MacApplication::isAperture()
-{
-    static bool isAperture = applicationBundleIsEqualTo("com.apple.Aperture"_s);
-    return isAperture;
-}
```
Comment (brentfulgham) [inline Source/WebCore/platform/cocoa/RuntimeApplicationChecksCocoa.mm:195]: Indeed, Aperture is not supported on modern macOS.

File: Source/WebKitLegacy/mac/WebCoreSupport/WebEditorClient.mm [hunk 2]
```diff
@@ -356,10 +356,6 @@ static void updateFontPanel(WebView *webView)
     }
 
 #if !PLATFORM(IOS_FAMILY)
-    // FIXME: This quirk is needed due to <rdar://problem/5009625> - We can phase it out once Aperture can adopt the new behavior on their end
-    if (!WebKitLinkedOnOrAfter(WEBKIT_FIRST_VERSION_WITHOUT_APERTURE_QUIRK) && [[[NSBundle mainBundle] bundleIdentifier] isEqualToString:@"com.apple.Aperture"])
```
Comment (brentfulgham) [inline Source/WebKitLegacy/mac/WebCoreSupport/WebEditorClient.mm:360]: Please delete the `#define WEBKIT_FIRST_VERSION_WITHOUT_APERTURE_QUIRK` in `Source/WebKitLegacy/Misc/WebKitVersionChecks.h`, too.

File: Source/WebCore/Modules/geolocation/Geolocation.cpp [hunk 3]
```diff
@@ -353,15 +353,6 @@ static void logError(const String& target, const bool isSecure, const bool isMix
...
 }
-
-// FIXME: remove this function when rdar://problem/32137821 is fixed.
-static bool isRequestFromIBooks()
-{
-#if PLATFORM(COCOA)
-    return CocoaApplication::isIBooks();
-#endif
-    return false;
-}
```
Comment (brentfulgham) [inline Source/WebCore/Modules/geolocation/Geolocation.cpp:364]: The underlying Radar has been fixed, so it is safe to remove this.

File: Source/WebCore/platform/cocoa/RuntimeApplicationChecksCocoa.mm [hunk 4]
```diff
@@ -140,12 +140,6 @@ static bool applicationBundleIsEqualTo(const String& bundleIdentifierString)
     return isAppleMail;
 }
 
-bool MacApplication::isITunes()
-{
-    static bool isITunes = applicationBundleIsEqualTo("com.apple.iTunes"_s);
-    return isITunes;
```
Comment (brentfulgham) [inline Source/WebCore/platform/cocoa/RuntimeApplicationChecksCocoa.mm:146]: iTunes is indeed unsupported on modern macOS, so this is safe to remove.

File: Source/WebCore/platform/cocoa/RuntimeApplicationChecksCocoa.mm [hunk 5]
```diff
@@ -158,12 +152,6 @@ static bool applicationBundleIsEqualTo(const String& bundleIdentifierString)
     return isAdobeInstaller;
 }
 
-bool MacApplication::isAOLInstantMessenger()
-{
-    static bool isAOLInstantMessenger = applicationBundleIsEqualTo("com.aol.aim.desktop"_s);
```
Comment (brentfulgham) [inline Source/WebCore/platform/cocoa/RuntimeApplicationChecksCocoa.mm:163]: This application has been decommissioned, and any modern variant is unlikely to need this quirk.

**Review by brentfulgham — APPROVED** [review]
This looks like a great cleanup -- thank you! Could you please also remove the `#define WEBKIT_FIRST_VERSION_WITHOUT_APERTURE_QUIRK` in `Source/WebKitLegacy/Misc/WebKitVersionChecks.h` for the complete cleanup?

**Comment by AtariDreams** [comment]
@brentfulgham Done!

**Pull request merged**